pkg/collector/reconcile/config_replace.go

```diff
@@ -71,13 +77,18 @@ func ReplaceConfig(instance v1alpha1.OpenTelemetryCollector) (string, error) {
 		}
 	}
 
-	updPromCfgMap := make(map[string]interface{})
-	if err := mapstructure.Decode(cfg, &updPromCfgMap); err != nil {
-		return "", err
+	if featuregate.EnableTargetAllocatorRewrite.IsEnabled() {
+		cfg.TargetAllocConfig = &targetAllocator{
+			Endpoint:    fmt.Sprintf("http://%s:80", naming.TAService(instance)),
+			Interval:    30 * time.Second,
+			CollectorID: "${POD_NAME}",
+		}
+		// we don't need the scrape configs here anymore with target allocator enabled
+		cfg.PromConfig.ScrapeConfigs = []*promconfig.ScrapeConfig{}
```

Comment:
I actually think the collector may fail to startup if it doesn't have any scrape configs set, at least that was the case a few months ago...

Reply:
oh ha, nvm i fixed this a few months ago ([link](https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/21f36f87fc1d2472adbb0a663f3bf6517701ff5a#diff-c335f316da4cfd1d6feec64b950de5600d871572b350b1fadfe411c54343e073R161-R162))